### ISSUE TITLE
`unnecessary_nesting_linter()` can be ambiguous in complex nesting statements

### ISSUE THREAD

**Issue by IndrajeetPatil** [r-lib/lintr#1891]
``` r
'if (a) {
  if (b) {
    if (c) {
      message("hi")
    }
    t <- 1L
  }
}' -> code

library(lintr)

lint(
  text = code,
  linters = unnecessary_nesting_linter()
)
#> <text>:2:3: warning: [unnecessary_nesting_linter] Don't use nested `if` statements, where a single `if` with the combined conditional expression will do. For example, instead of `if (x) { if (y) { ... }}`, use `if (x && y) { ... }`.
#>   if (b) {
#>   ^~~~~~~~
```

<sup>Created on 2023-01-06 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

But the suggested modification is not equivalent to the original code:

```r
if (a) {
  if (b && c) {
    message("hi")
    t <- 1L
  }
}
```

**Comment by AshesITR** [comment]
The lint seems to (correctly) land in the inner if:

```r
if (a && b) {
  if (c) {
    message("hi")
  }
  t <- 1L
}
```

**Comment by IndrajeetPatil** [comment]
Ah, yes, you are right! For some reason, I reasoned downward (`b && c`) from the lint location, when the upward (`a && b`) is expected.

I wonder if I will be the last person who will make this mistake, and if there is anything we can include in the lint message to anticipate and foreclose this possibility?

**Comment by IndrajeetPatil** [comment]
Yes, I would find that to be a clearer message to understand.
